coding/writer.hpp

```diff
@@ -61,8 +61,11 @@ template <typename ContainerT> class MemWriter : public Writer
 
   void Write(void const * p, size_t size) override
   {
+    if (size == 0)
+        return;
+
     auto freeSize = static_cast<intptr_t>(m_Data.size() - m_Pos);
```

Comment:
Should size and pos be casted to intptr before the subtraction?

Reply:
Not needed - `m_Data.size()` is always greater or equal than `m_Pos`.

Reply:
- Hm, if so, what is the reason in ```if (freeSize < 0)``` logic? We can do Seek in any position according to the code.
- It is not reproducible in our situation, but the current code is ill-formed in a very corner case: https://coliru.stacked-crooked.com/a/0ffdb84c8e972040

Reply:
Of course, no need to modify this logic, because size and m_Pos are fit into intptr_t.